### PR TITLE
deepprofiler project processing features

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
@gwaybio @d33bs 

Here is the PR for normalizing and feature selection of DeepProfiler features.

Please feel free to review this and I will perform analysis after :smile:

### CODE REVIEW

File: 4_processing_features/extract_sc_features_dp.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+# ### Feature selection from normalized data
+
+# In[9]:
+
+
+feature_select_ops = [
+    "variance_threshold",
+    "correlation_threshold",
+    "blocklist",
```
Comment (gwaybio) [inline 4_processing_features/extract_sc_features_dp.py]: Blocklist will only work with CP features. You can remove!

Comment (gwaybio) [reply 4_processing_features/extract_sc_features_dp.py]: (Please remove in both nuc and cyto)

**Review by gwaybio — CHANGES_REQUESTED** [review]
One minor comment described below.

One major comment is the separation of nuc and cyto features. Can you remind me why we needed to split these? Is there a way to combine them into a single dataset instead of splitting them out?

**Comment by jenna-tomkinson** [comment]
@gwaybio 

This separation (also seen in [cell-health](https://github.com/jenna-tomkinson/cell-health-data/tree/master/2.extract-features)) is due to the Cellpose center x,y coords for each object is different. As well, since DeepProfiler creates a box around these coords, the parameters for the box size is different between the objects. This is what leads us to creating a DeepProfiler project for each object.

As for making them a single dataset, we could do a merging function (like what is being made in idrstream) to merge the features from each project to the single cells based on the center x,y coordinates.

What are your thoughts on this?

File: 4_processing_features/extract_sc_features_dp.py (hunk 1 above)
Comment (jenna-tomkinson) [reply 4_processing_features/extract_sc_features_dp.py]: 👍 Done!

**Comment by gwaybio** [comment]
right! Glad to know the merging function is going to be necessary in multiple projects - we're definitely going to want modularity!

Our univariate tests should be fine with this split, but our multivariate tests will be impacted.

Can you create an issue describing the need to link them, so that we don't forget in the future?

For now, we can perform the univariate tests separately and apply the multivariate tests on each individual data set (nuc and cyto) separately. We'll perform a combined multivariate test once we have a linking function.

**Comment by jenna-tomkinson** [comment]
@gwaybio 

Sounds good! I will make that issue now! Are we okay to merge this PR through after that change to feature selection?

**Review by gwaybio — APPROVED** [review]
Yep. LGTM

File: 4_processing_features/extract_sc_features_dp.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+# # Process single cell morphology features for DeepProfiler readouts
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
```
Comment (d33bs) [inline 4_processing_features/extract_sc_features_dp.py]: Consider removing this pandas import if it is not needed.

File: 4_processing_features/4.extract_sc_features.sh [hunk 3]
```diff
@@ -1,3 +1,6 @@
 #!/bin/bash
-jupyter nbconvert --to python extract_single_cell_features.ipynb
-python extract_single_cell_features.py
+jupyter nbconvert --to python extract_sc_features_cp.ipynb
+python extract_sc_features_cp.py
+
+jupyter nbconvert --to python extract_sc_features_dp.ipynb
```
Comment (d33bs) [inline 4_processing_features/4.extract_sc_features.sh]: Noticing the multiple explicit calls to `nbconvert`, would it be helpful to consider one call for all/any jupyter notebooks within this directory?

**Review by d33bs — APPROVED** [review]
Nice job - I left a couple minor comments but LGTM!

File: 4_processing_features/4.extract_sc_features.sh (hunk 3 above)
Comment (jenna-tomkinson) [reply 4_processing_features/4.extract_sc_features.sh]: I was actually thinking about that when I was doing it but I didn't decided to. I don't know why! I will implement this!

Comment (d33bs) [reply 4_processing_features/4.extract_sc_features.sh]: Sounds good!

File: 4_processing_features/extract_sc_features_dp.py (hunk 2 above)
Comment (jenna-tomkinson) [reply 4_processing_features/extract_sc_features_dp.py]: 👍 Great catch ty!

**Pull request merged**